vectordatasource/transform.py

```diff
@@ -2849,6 +2850,126 @@ def merge_features(ctx):
     return layer
 
 
+def quantize_val(val, step):
+    result = int(5 * round(val / float(step)))
+    return result
+
+
+def quantize_height_round_nearest_5_meters(height):
+    return quantize_val(height, 5)
+
+
+def quantize_height_round_nearest_10_meters(height):
+    return quantize_val(height, 10)
+
+
+def quantize_height_round_nearest_meter(height):
+    return round(height)
+
+
+def merge_building_features(ctx):
+    zoom = ctx.tile_coord.zoom
+    source_layer = ctx.params.get('source_layer')
+    start_zoom = ctx.params.get('start_zoom', 0)
+    end_zoom = ctx.params.get('end_zoom')
+    drop = ctx.params.get('drop')
+    exclusions = ctx.params.get('exclude')
+
+    assert source_layer, 'merge_building_features: missing source layer'
+
+    if zoom < start_zoom:
+        return None
+    if end_zoom is not None and zoom > end_zoom:
+        return None
+
+    layer = _find_layer(ctx.feature_layers, 'buildings')
+    if layer is None:
+        return None
+
+    quantize_height_fn = None
+    quantize_cfg = ctx.params.get('quantize')
+    if quantize_cfg:
+        quantize_fn_dotted_name = quantize_cfg.get(zoom)
+        if quantize_fn_dotted_name:
+            quantize_height_fn = resolve(quantize_fn_dotted_name)
+
+    features_by_property = {}
+    skipped_features = []
+    for feature in layer['features']:
+        shape, props, fid = feature
+        dims = _geom_dimensions(shape)
+        if dims != _POLYGON_DIMENSION:
+            skipped_features.append(feature)
+            continue
+
+        if exclusions:
+            for prop in exclusions:
+                if prop in props:
+                    skipped_features.append(feature)
+                    continue
+
+        orig_props = props.copy()
+
+        p_id = props.pop('id', None)
+        # also drop building properties that we won't want to consider
+        # for merging. area and volume will be re-calculated afterwards
+        props.pop('area', None)
+        props.pop('volume', None)
+
+        if drop:
+            for prop in drop:
+                props.pop(prop, None)
+
+        props.pop('addr_housenumber', None)
+        props.pop('addr_street', None)
```

Comment:
These are in the `drop` list as currently configured, and would we want them dropped if not configured?